File: backend/app/schemas/project.py
from typing import Optional, Dict, Any
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class ProjectBase(BaseModel):
    jira_key: str
    name: str
    description: Optional[str] = None
    budget: Optional[float] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    status: str = "active"
    meta: Optional[Dict[str, Any]] = None


class ProjectCreate(ProjectBase):
    owner_id: int


class ProjectUpdate(BaseModel):
    jira_key: Optional[str] = None
    name: Optional[str] = None
    description: Optional[str] = None
    budget: Optional[float] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    status: Optional[str] = None
    meta: Optional[Dict[str, Any]] = None


class ProjectInDB(ProjectBase):
    id: int
    owner_id: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class Project(ProjectInDB):
    pass


class ProjectWithStats(Project):
    total_tasks: int = 0
    completed_tasks: int = 0
    in_progress_tasks: int = 0
    total_estimate_hours: float = 0
    total_spent_hours: float = 0
    completion_percentage: float = 0
    velocity: Optional[float] = None

File: frontend/src/services/api/types/project.ts
// project domain API types (split from types.ts, wave E5).

import type { RepositoryInfo } from './integrations';



export interface Project {
  id: number;
  jira_key: string;
  name: string;
  description?: string;
  status: string;
  budget?: number;
  spent_budget?: number;
  start_date?: string | null;
  end_date?: string | null;
  owner_id?: number | null;
  meta?: {
    last_sync_at?: string;
    issues_count?: number;
    [key: string]: unknown;
  };
  created_at?: string;
  updated_at?: string | null;
  // Stats (when returned by detail endpoint)
  total_tasks?: number;
  completed_tasks?: number;
  in_progress_tasks?: number;
  total_estimate_hours?: number;
  total_spent_hours?: number;
  completion_percentage?: number;
  velocity?: number | null;
  // Quality thresholds for PR gates
  quality_thresholds?: {
    min_line?: number;
    min_branch?: number;
  };
}

export interface ProjectRepositoryLink {
  id: number;
  project_id: number;
  repository_id: number;
  is_primary: boolean;
  created_at: string;
  updated_at?: string | null;
  repository: RepositoryInfo;
}
